local.tf

```diff
@@ -6,6 +6,7 @@ locals {
   cluster_security_group_id = "${coalesce(join("", aws_security_group.cluster.*.id), var.cluster_security_group_id)}"
 
   worker_security_group_id = "${coalesce(join("", aws_security_group.workers.*.id), var.worker_security_group_id)}"
+  default_iam_role_id      = "${element(concat(aws_iam_role.workers.*.id, list("")), 0)}"
```

Comment:
I am confused as to what this is supposed to accomplish - 
1. `aws_iam_role.workers` doesn't have `count` set, so the splat operator [shouldn't work here](https://www.terraform.io/docs/configuration/interpolation.html#attributes-of-other-resources) as far as I know:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/3b546a9c1bc9fab24b3bd0ca5ec30b5defed5e50/workers.tf#L95
(if the splat operator works on single resources, that's fine)

2. aws_iam_role.workers only depends on the cluster name, so it shouldn't ever be null, so why do we need this in the first place?

Reply:
Issue report commented as well, this is to support all `destroy` operations, as locals behave similar to outputs, and cause errors when the resources don't exist.

Trying to use the same element/concat pattern as typically applied to output variables.

Reply:
Oh, it's for destroy operations 🤦‍♂️ 
I don't recall ever hitting that race condition during a teardown, but then I've only torn this version down a handful of times. Still surprised the splat operator works, but this makes a lot more sense to me now!